src/rpc/net.cpp

```diff
@@ -1014,23 +1015,58 @@ static RPCHelpMan sendmsgtopeer()
-                                       {
-                                               {RPCResult::Type::NUM, "new", "number of addresses in the new table, which represent potential peers the node has discovered but hasn't yet successfully connected to."},
-                                               {RPCResult::Type::NUM, "tried", "number of addresses in the tried table, which represent peers the node has successfully connected to in the past."},
-                                               {RPCResult::Type::NUM, "total", "total number of addresses in both new/tried tables"},
-                                       }},
-                              }
+                      {
+                              {"verbose", RPCArg::Type::BOOL, RPCArg::Default{false}, "True for a list of the individual address manager entries, false for address counts only"},
+                      },
+                      {
+                              RPCResult{"for verbose = false",
+                                      RPCResult::Type::OBJ_DYN, "", "json object with network type as keys",
+                                      {
+                                              {RPCResult::Type::OBJ, "network", "the network (" + Join(GetNetworkNames(), ", ") + ")",
+                                               {
+                                                       {RPCResult::Type::NUM, "new", "number of addresses in the new table, which represent potential peers the node has discovered but hasn't yet successfully connected to."},
+                                                       {RPCResult::Type::NUM, "tried", "number of addresses in the tried table, which represent peers the node has successfully connected to in the past."},
+                                                       {RPCResult::Type::NUM, "total", "total number of addresses in both new/tried tables"},
+                                               }},
+                                      },
+                              },
+                              RPCResult{"for verbose = true",
+                                      RPCResult::Type::OBJ_DYN, "", "json object with network type, new_table and tried_table as keys",
+                                      {
+                                              {RPCResult::Type::ELISION, "", "The same output as verbose = false"},
+                                              {RPCResult::Type::ARR, "table", "list of addresses in a address manager table ( new_table, tried_table )",
+                                              {
+                                                {RPCResult::Type::OBJ, "", "an address manager table entry",
+                                                    {
+                                                        {RPCResult::Type::STR, "address", "the address"},
+                                                        {RPCResult::Type::NUM, "services", "the services the node might support"},
```

Comment:
Since this is a hidden/test-only RPC I don't think we need services and service_names as we have in `getpeerinfo`. If people disagree, I'd be happy to add it.